plugins/azure-devops/src/components/BuildTable/BuildTable.tsx

```diff
@@ -113,6 +115,43 @@ export const getBuildStateComponent = (
   }
 };
 
+export const getBuildDuration = (startTime?: Date, finishTime?: Date) => {
+  if (!startTime || (!startTime && !finishTime)) {
+    return '';
+  }
+
+  const start = DateTime.fromISO(startTime.toString());
+  const finish = finishTime
+    ? DateTime.fromISO(finishTime.toString())
+    : DateTime.now();
+
+  const formatted = Interval.fromDateTimes(start, finish)
+    .toDuration()
+    .valueOf();
+
+  const shortEnglishHumanizer = humanizeDuration.humanizer({
+    language: 'shortEn',
+    languages: {
+      shortEn: {
+        y: () => 'y',
+        mo: () => 'mo',
+        w: () => 'w',
+        d: () => 'd',
+        h: () => 'h',
+        m: () => 'm',
+        s: () => 's',
+        ms: () => 'ms',
+      },
+    },
+  });
+
+  return shortEnglishHumanizer(formatted, {
+    largest: 2,
+    round: true,
+    spacer: '',
+  });
+};
```

Comment:
This can probably be a common function in a seperate file as `getDurationFromDates` or something similar.

Reply:
Yes, I waffled on doing that. I've extracted it and the related tests